kivy/_clock.pyx

```diff
@@ -402,6 +410,10 @@ cdef class CyClockBase(object):
         self._next_event = self._cap_event = None
         self._lock_release()
 
+        while self._del_queue:
+            callback = self._del_queue.pop()
+            callback()
+
```

Comment:
A slightly more performant approach is:

```py
callbacks = self._del_queue[:]
del self._del_queue[:len(callbacks)]
for callback in callbacks:
    callback()
```

The reason is that it's cheaper than popping one callback at a time. Notice how it makes sure it only deletes the callbacks it'll process in this frame (this wasn't an issue with your code though).